pkg/apis/operator/v1/types.go

```diff
@@ -369,18 +369,67 @@ func (cp CNIPluginType) String() string {
 	return string(cp)
 }
 
+type IPAMPluginType string
+
+const (
+	IPAMPluginCalico    IPAMPluginType = "Calico"
+	IPAMPluginHostLocal IPAMPluginType = "HostLocal"
+	IPAMPluginAmazonVPC IPAMPluginType = "AmazonVPC"
+	IPAMPluginAzureVNET IPAMPluginType = "AzureVNET"
+)
+
+var IPAMPluginTypes []IPAMPluginType = []IPAMPluginType{
+	IPAMPluginCalico,
+	IPAMPluginHostLocal,
+	IPAMPluginAmazonVPC,
+	IPAMPluginAzureVNET,
+}
+
+var IPAMPluginTypesString []string = []string{
+	IPAMPluginCalico.String(),
+	IPAMPluginHostLocal.String(),
+	IPAMPluginAmazonVPC.String(),
+	IPAMPluginAzureVNET.String(),
+}
+
+func (cp IPAMPluginType) String() string {
+	return string(cp)
+}
+
+type IPAMSpec struct {
+	// Specifies the IPAM plugin that will be used in the Calico or Calico Enterprise installation.
+	// * For CNI Plugin Calico, this field defaults to Calico.
+	// * For CNI Plugin GKE, this field defaults to HostLocal.
+	// * For CNI Plugin AzureVNET, this field defaults to AzureVNET.
+	// * For CNI Plugin AmazonVPC, this field defaults to AmazonVPC.
+	//
+	// The IPAM plugin is installed and configured only if the CNI plugin is set to Calico,
+	// for all other values of the CNI plugin the plugin binaries and CNI config is a dependency
+	// that is expected to be installed separately.
+	//
+	// Default: Calico
+	// +kubebuilder:validation:Enum=Calico;HostLocal;AmazonVPC;AzureVNET
+	Type IPAMPluginType `json:"type"`
+}
+
 type CNISpec struct {
 	// Specifies the CNI plugin that will be used in the Calico or Calico Enterprise installation.
-	// For KubernetesProvider GKE, this field defaults to GKE. For KubernetesProvider AKE, this field
-	// defaults to AzureVNET. For KubernetesProvider EKS, this field defaults to AmazonVPC. For all other
-	// KubernetesProviders this field defaults to Calico.
+	// * For KubernetesProvider GKE, this field defaults to GKE.
+	// * For KubernetesProvider AKS, this field defaults to AzureVNET.
+	// * For KubernetesProvider EKS, this field defaults to AmazonVPC.
+	// * For all other KubernetesProviders this field defaults to Calico.
+	//
 	// For the value Calico, the CNI plugin binaries and CNI config will be installed as part of deployment,
 	// for all other values the CNI plugin binaries and CNI config is a dependency that is expected
 	// to be installed separately.
 	//
 	// Default: Calico
 	// +kubebuilder:validation:Enum=Calico;GKE;AmazonVPC;AzureVNET
 	Type CNIPluginType `json:"type"`
+
+	// IPAM specifies the IPAM that will be used in the Calico or Calico Enterprise installation.
```

Comment:
This is a bit self-referential (IPAM specifies the IPAM).

Maybe "IPAM configures pod IP address management"